File: app/src/main/java/org/test/kotlin/demo/api/dto/SearchDTO.java
package org.test.kotlin.demo.api.dto;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class SearchDTO<T> {

    @SerializedName("items")
    private List<T> items;

    @SerializedName("total_count")
    private int totalCount;

    @SerializedName("incomplete_results")
    private boolean incompleteResults;

    public List<T> getItems() {
        return items;
    }

    public void setItems(List<T> items) {
        this.items = items;
    }

    public int getTotalCount() {
        return totalCount;
    }

    public void setTotalCount(int totalCount) {
        this.totalCount = totalCount;
    }

    public boolean isIncompleteResults() {
        return incompleteResults;
    }

    public void setIncompleteResults(boolean incompleteResults) {
        this.incompleteResults = incompleteResults;
    }

}

File: app/src/main/java/org/test/kotlin/demo/api/dto/SearchDTO.kt
package org.test.kotlin.demo.api.dto

import com.google.gson.annotations.SerializedName

data class SearchDTO<T>(
        @SerializedName("items") val items: List<T>,
        @SerializedName("total_count") val totalCount: Int,
        @SerializedName("incomplete_results") val isIncompleteResults: Boolean
)
